app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.trainingintive.MainActivity
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class SplashViewModel : ViewModel() {
+
+    init {
+        start()
+    }
+
+    private val _navigator = MutableLiveData<Class<out AppCompatActivity?>>()
+    val navigator: LiveData<Class<out AppCompatActivity?>> = _navigator
```

Comment:
You have to create separate class which will handle navigation events out of the ViewModel scope. It should be a layer of abstraction over Android. Think about cases when your navigator will have to handle navigation to separate activity (by startActivity/ForResult) or by using navigation component (by navController). Maybe you will also need to show Dialog. All cases should be handled by your navigator class. 
To send action from ViewModel to your navigator you can use kotlin `sealed class` to create event which will be handled by your navigator.